zmq.go

```diff
@@ -404,22 +445,25 @@ func Subscribe(addr string, topics []string, timeout time.Duration) (*Conn, erro
 	return c, nil
 }
 
-// Receive a message from the publisher. It blocks until a new message is
-// received. If the connection times out and it was not explicitly terminated,
-// then a timeout error is returned. Otherwise, if it was explicitly terminated,
-// then io.EOF is returned.
-func (c *Conn) Receive() ([][]byte, error) {
-	messages, err := c.readMessage()
+// Receive a message from the publisher. Messages can be composed of multiple
+// sub-messages. They are read into the provided buffers. Each buffer should be
+// of sufficient length to successfully read messages. If none are provided,
+// then buffers will be allocated for each sub-message. It blocks until a new
+// message is received. If the connection times out and it was not explicitly
+// terminated, then a timeout error is returned. Otherwise, if it was explicitly
+// terminated, then io.EOF is returned.
+func (c *Conn) Receive(bufs [][]byte) error {
```

Comment:
Doesn't appear to allow the caller to not allocate any buffers themselves, yet still get the received message as a return value. In order to emulate this, we need to restore the prior return value, and then just return `bufs` again.

Reply:
Should be fixed now.